wpilibNewCommands/src/main/native/cpp/frc2/command/Subsystem.cpp

```diff
@@ -16,6 +18,10 @@ void Subsystem::Periodic() {}
 
 void Subsystem::SimulationPeriodic() {}
 
+std::string Subsystem::GetName() const {
+  return wpi::GetTypeName(*this);
```

Comment:
My quick testing showed that it resolves to the class name, but feel free to try to break it. 

The current behavior of SubsystemBase.GetName() *does* resolve to "class frc2::SubsystemBase", because `GetTypeName` is called in the SubsystemBase constructor. I'm not sure there's a way to get around this cleanly.